### PR TITLE
anonymize to delete();

### LINKED ISSUE #5440: de-anonymise UserProfile
from irc

> — @eviljeff notes three different references to 'anonymous' in UserProfile - is_anonymous; has_anonymous_username; anonymize (not logged in; no username set; deleted) :/
> — @eviljeff tries to come up with non-stupid synonyms

I'm thinking of changing `def anonymize` to override `delete` as we never actually delete.  And it's how we do soft-delete on other models.

`is_anonymous` is defined on AbstractBaseUser https://docs.djangoproject.com/en/1.8/topics/auth/customizing/#django.contrib.auth.models.AbstractBaseUser.is_anonymous so we need to keep that one.

anonymous usernames just need renaming.

### PR DESCRIPTION
**Pull request by eviljeff**
Sort of fixes #5440.  

Doesn't rename anonymous_username and similar -  partly because I couldn't think of a good alternative name; partly because we use 'anonymous' in the username so it might be (more) confusing if the methods setting that are called something different.

### CODE REVIEW

File: src/olympia/users/models.py [hunk 1]
```diff
@@ -177,9 +177,6 @@ def has_module_perms(self, app_label):
 
     backend = 'django.contrib.auth.backends.ModelBackend'
 
-    def is_anonymous(self):
```
Comment (diox) [inline src/olympia/users/models.py]: Ah, I knew we had our own implementation, I just didn't know django had it too :p

Comment (eviljeff) [inline src/olympia/users/models.py]: Yep.  I thought it'd be clearer if we didn't duplicate it in our subclass.

File: src/olympia/zadmin/tests/test_views.py [hunk 2]
```diff
@@ -1978,3 +1978,21 @@ def test_unprivileged_user(self):
...
+        self.grant_permission(self.user, '*:*')
+        self.login(self.user)
+
+    def test_delete_does_hard_delete(self):
+        user_to_delete = user_factory()
+        user_to_delete_pk = user_to_delete.pk
+        url = reverse('admin:users_userprofile_delete',
+                      args=[user_to_delete.pk])
+        response = self.client.post(url, data={'post': 'yes'}, follow=True)
+        assert response.status_code == 200
```
Comment (diox) [inline src/olympia/zadmin/tests/test_views.py]: It doesn't redirect ? I'm slightly surprised.

Comment (eviljeff) [inline src/olympia/zadmin/tests/test_views.py]: `follow=True`

Comment (diox) [inline src/olympia/zadmin/tests/test_views.py]: I need more sleep...

**Pull request merged**